tests/backup/backup_restore_basic_test.go

```diff
@@ -1442,3 +1442,220 @@ var _ = Describe("{BackupMultipleNsWithSameLabel}", func() {
+			log.FailOnError(err, fmt.Sprintf("Fetching [%s] cluster status", SourceClusterName))
+			dash.VerifyFatal(clusterStatus, api.ClusterInfo_StatusInfo_Online, fmt.Sprintf("Verifying if [%s] cluster is online", SourceClusterName))
+			clusterUid, err = Inst().Backup.GetClusterUID(ctx, orgID, SourceClusterName)
+			dash.VerifyFatal(err, nil, fmt.Sprintf("Fetching [%s] cluster uid", SourceClusterName))
+		})
+
+		Step("Taking backup of applications", func() {
+			log.InfoD("Taking backup of applications")
+			var sem = make(chan struct{}, numberOfSimultaneousBackups)
+			var wg sync.WaitGroup
+			ctx, err := backup.GetAdminCtxFromSecret()
+			log.FailOnError(err, "Fetching px-central-admin ctx")
+			log.InfoD("Taking %d backups", numberOfBackups)
+			for backupLocationUID, backupLocationName := range backupLocationMap {
+				for _, namespace := range bkpNamespaces {
+					for i := 0; i < numberOfBackups; i++ {
+						sem <- struct{}{}
+						time.Sleep(timeBetweenConsecutiveBackups)
+						backupName := fmt.Sprintf("%s-%v", BackupNamePrefix, time.Now().Unix())
+						backupNames = append(backupNames, backupName)
+						wg.Add(1)
+						go func(backupName string) {
+							defer GinkgoRecover()
+							defer wg.Done()
+							defer func() { <-sem }()
+							err = CreateBackup(backupName, SourceClusterName, backupLocationName, backupLocationUID, []string{namespace},
+								labelSelectors, orgID, clusterUid, "", "", "", "", ctx)
+							dash.VerifyFatal(err, nil, fmt.Sprintf("Verifying backup creation: %s", backupName))
+						}(backupName)
+					}
+					wg.Wait()
+				}
+			}
+			log.Infof("List of backups - %v", backupNames)
+		})
+
+		Step("Remove the backup locations where backups were taken", func() {
+			log.InfoD("Remove backup locations where backups were taken")
+			// Issue a remove backup location call
+			for backupLocationUID, customBackupLocationName = range backupLocationMap {
+				err := DeleteBackupLocation(customBackupLocationName, backupLocationUID, orgID, false)
+				dash.VerifySafely(err, nil, fmt.Sprintf("Deleting backup location %s", customBackupLocationName))
+
+				// Wait until backup location is removed
+				backupLocationDeleteStatusCheck := func() (interface{}, bool, error) {
+					ctx, err := backup.GetAdminCtxFromSecret()
+					log.FailOnError(err, "Fetching px-central-admin ctx")
+					status, err := IsBackupLocationPresent(customBackupLocationName, ctx, orgID)
+					if err != nil {
+						return "", true, fmt.Errorf("backup location %s still present with error %v", customBackupLocationName, err)
+					}
+					if status == true {
+						return "", true, fmt.Errorf("backup location %s is not deleted yet", customBackupLocationName)
+					}
+					return "", false, nil
+				}
+				_, err = task.DoRetryWithTimeout(backupLocationDeleteStatusCheck, 3*time.Minute, 30*time.Second)
+				dash.VerifySafely(err, nil, fmt.Sprintf("Deleting backup location %s", customBackupLocationName))
```

Comment:
Shouldn't this be `dash.VerifyFatal()` as there's no point in moving on with the test if we failed to delete a backup location.

Reply:
What's the problem with `dash.VerifyFatal()`?

Reply:
@kshithijiyer-px If you don't want to proceed with the test case if this task fails then we have to use `dash.VerifyFatal()`
For tasks in cleanup where we dont want to hold up the remaining cleanup tasks because the first cleanup task fails we use `dash.VerifySafely()`

Reply:
the problem is that it continues with the code, even if it fails to delete a backup location. But if we fail to delete, then we can't actually test if creating the location again, will still have the backups (after sync), right?

Reply:
Done